libs/portal-integration-angular/src/lib/services/portal-dialog.service.ts

```diff
@@ -1,8 +1,424 @@
+
+/**
+ * Object containing key for translation with parameters object for translation
+ *
+ * @example
+ * ## TranslationModule content
+ * ```typescript
+ * const translations = {
+ *   MY_KEY = 'text with parameter value = {{value}}',
+ * }
+ * ```
+ *
+ * ## TranslationKeyWithParameters declaration
+ * ```
+ * // will be translated into
+ * // text with parameter value = hello
+ * const myKey: TranslationKeyWithParameters = {
+ *   key: 'MY_KEY',
+ *   parameters: {
+ *     value: 'hello',
+ *   },
+ * }
+ * ```
+ */
+type TranslationKeyWithParameters = { key: string; parameters: Record<string, unknown> }
+/**
+ * String with key to translation or {@link TranslationKeyWithParameters} object. If provided string cannot be translated it will be displayed as is.
+ */
+type TranslationKey = string | TranslationKeyWithParameters
+/**
+ * Object containing message of type {@link TranslationKey} and icon to be displayed along the message.
+ *
+ * @example
+ * DialogMessage with TranslationKey will display 'text with parameter value = hello' and question mark icon
+ *
+ * ## TranslationModule content
+ * ```
+ * const translations = {
+ *   MY_KEY = 'text with parameter value = {{value}}',
+ * }
+ * ```
+ *
+ * ## DialogMessage declaration
+ * ```
+ * const myDialogMessage: DialogMessage = {
+ *   message: {
+ *     key: 'MY_KEY',
+ *     parameters: {
+ *       value = 'hello',
+ *     },
+ *   },
+ *   icon: 'pi pi-question'
+ * }
+ * ```
+ */
+type DialogMessage = { message: TranslationKey; icon: string }
+
+/**
+ * Implement via component class to be displayed by {@link PortalDialogService.openDialog}
+ *
+ * Use if you want {@link PortalDialogService.openDialog} to return state of displayed component's current dialogResult value alongside the clicked button.
+ *
+ * @example
+ * Display component implementing DialogResult<string> and react on the returned value
+ *
+ * ## Component declaration
+ * ```
+ * ⁣@Component({template: `<div>
+ * <input (change)="onInputChange($event)">
+ * </div>`})
+ * export class MyInputComponent implements DialogResult<string> {
+ *   dialogResult: string = ''
+ *
+ *   onInputChange(event: any) {
+ *     dialogResult = event.target.value
+ *   }
+ * }
+ * ```
+ *
+ * ## PortalDialogService call
+ * ```
+ * portalDialogService.openDialog(title, { type: MyInputComponent }, primaryButton, ...).subscribe((result: DialogState<string>) => {
+ * // result.value === MyInputComponent.dialogResult (during button click)
+ * // behavior when dialog closes
+ * })
+ * ```
+ *
+ */
+export interface DialogResult<T> {
+  dialogResult: T
+}
+/**
+ * Implement via component class to be displayed by {@link PortalDialogService.openDialog}
+ *
+ * Use to control the state of the primary button (disabled or enabled). Whenever your component wants to disable/enable primary button it should emit boolean equal to whether primary button should be enabled.
+ */
```

Comment:
You should add that if you implement this interface the button is displayed till it is enabled via the emitter